webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -278,4 +303,161 @@ describe('TasksComponent', () => {
+      const expectedTasks =
+        [
+          {
+            _id: 'e1',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+              'CHW Bettys Area'
+            ],
+            overdue: true,
+            owner: 'a',
+          },
+          {
+            _id: 'e2',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+            ],
+            overdue: true,
+            owner: 'b',
+          }
+        ];
+      userContactService.get.resolves(bettysContactDoc);
+      sessionService.isOnlineOnly.returns(true);
+      rulesEngineService.fetchTaskDocsForAllContacts.resolves(taskDocs);
+      lineageModelGeneratorService.reportSubjects.resolves(taskLineages);
+      await new Promise(resolve => {
+        sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+        getComponent();
+      });
+      expect(component.currentLevel).to.be.undefined;
+      expect((<any>TasksActions.prototype.setTasksList).args).to.deep.equal([[expectedTasks]]);
+    });
+
+    it('it should not change the reports lineages if the connected user is offline with unrelated lineage', async() => {
+      const expectedTasks =
+        [
+          {
+            _id: 'e1',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+              'CHW Bettys Area'
+            ],
+            overdue: true,
+            owner: 'a',
+          },
+          {
+            _id: 'e2',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+            ],
+            overdue: true,
+            owner: 'b',
+          }
+        ];
+      userContactService.get.resolves(userContactDoc);
+      sessionService.isOnlineOnly.returns(false);
+      rulesEngineService.fetchTaskDocsForAllContacts.resolves(taskDocs);
+      lineageModelGeneratorService.reportSubjects.resolves(taskLineages);
+      await new Promise(resolve => {
+        sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+        getComponent();
+      });
+      const tasks = await (<any>TasksActions.prototype.setTasksList).args;
+      expect(component.currentLevel).to.equal('parent');
+      expect(tasks).to.deep.equal([[expectedTasks]]);
+    });
+
+    it('it should update the reports lineages if the connected user is offline with related place', async() => {
```

Comment:
Targets! no reports :)

And space after async as it's reserved word.
```suggestion
    it('should update the tasks lineage if user is offline with related place to lineage', async () => {
```